components/system-broker/pkg/oauth/token_provider_test.go

```diff
@@ -0,0 +1,157 @@
+package oauth_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	httputils "github.com/kyma-incubator/compass/components/system-broker/pkg/http"
+	"github.com/kyma-incubator/compass/components/system-broker/pkg/oauth"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"io/ioutil"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+	"time"
+)
+
+func TestTokenProviderFromValue_GetAuthorizationToken(t *testing.T) {
+	tokenValue := "token"
+
+	testProvider := oauth.NewTokenProviderFromValue(tokenValue)
+	token, err := testProvider.GetAuthorizationToken(context.TODO())
+
+	require.NoError(t, err)
+	require.Equal(t, tokenValue, token.AccessToken)
+}
+
+func TestTokenProviderFromSecretTestSuite(t *testing.T) {
+	suite.Run(t, new(TokenProviderFromSecretTestSuite))
+}
+
+type TokenProviderFromSecretTestSuite struct {
+	suite.Suite
+
+	clientID      string
+	clientSecret  string
+	tokenEndpoint string
+	accessToken   string
+
+	config    *oauth.Config
+	k8sClient client.Client
+}
+
+func (suite *TokenProviderFromSecretTestSuite) SetupTest() {
```

Comment:
some tests (test files) are one a testify suite and others are not?

Reply:
We use testify suites when there is common before/after each logic. testify suites are just a wrapper of go sub-tests and there is no need of strong convention just a convenient way to separate before/after logic. That's why it is not always needed. There are test files which has standalone top-level tests and suites aswell.